server/src/main/java/org/cloudfoundry/identity/uaa/scim/vmidentity/VmidentityScimGroupProvisioning.java

```diff
@@ -139,71 +140,113 @@ public ScimGroup delete(String id, int version) {
     public List<ScimGroup> query(String filter, String sortBy, boolean ascending) {
-        logger.debug("group flter: " + filter);
-        return new ArrayList<ScimGroup>();
+        List<ScimGroup> groups = this.query(filter);
+
+        if (sortBy != null) {
+            sortScimGroups(groups, sortBy, ascending);
+        }
+
+        return groups;
     }
 
     @Override
     public int delete(String filter) {
-        throw new UnsupportedOperationException();
-    }
+        logger.debug("Filtering groups with query: " + filter);
 
-    @Override
-    public ScimGroupExternalMember mapExternalGroup(String groupId, String externalGroup, String origin)
-            throws ScimResourceNotFoundException, MemberAlreadyExistsException {
-        // TODO proper impl
-        ScimGroupExternalMember mem = new ScimGroupExternalMember(groupId, externalGroup);
-        mem.setOrigin(origin);
-        return mem;
+        try {
+            String tenant = VmidentityUtils.getTenantName(idmClient);
+            Set<Group> groups = idmClient.findGroupsByScimFilter(tenant, filter);
+            for (Group group : groups) {
+                idmClient.deletePrincipal(tenant, group.getId().getUPN());
+            }
+            return groups.size();
+        } catch (Exception e) {
+            logger.error("Unable to delete groups by filter '" + filter + "'", e);
+            throw new VmidentityDataAccessException("Unable to delete groups by filter: " + filter);
+        }
     }
 
-    @Override
-    public ScimGroupExternalMember unmapExternalGroup(String groupId, String externalGroup, String origin)
-            throws ScimResourceNotFoundException {
-        // TODO proper impl
-        ScimGroupExternalMember mem = new ScimGroupExternalMember(groupId, externalGroup);
-        mem.setOrigin(origin);
-        return mem;
+    static ScimGroup getScimGroup(Group idmGroup, String tenant, String systemDomain) {
+        String upn = VmidentityUtils.getPrincipalUpn(idmGroup.getId());
+        ScimGroup group = new ScimGroup(upn, upn, tenant);
+        group.setDescription(idmGroup.getDetail().getDescription());
+        ScimMeta meta = new ScimMeta();
+        meta.setVersion(1);
+        group.setMeta(meta);
+        return group;
     }
 
     @Override
-    public List<ScimGroupExternalMember> getExternalGroupMapsByGroupId(String groupId, String origin)
-            throws ScimResourceNotFoundException {
+    public int deleteByIdentityZone(String zoneId) {
         // TODO Auto-generated method stub
-        return new ArrayList<ScimGroupExternalMember>();
+        return 0;
     }
 
     @Override
-    public List<ScimGroupExternalMember> getExternalGroupMapsByExternalGroup(String externalGroup, String origin)
-            throws ScimResourceNotFoundException {
-        return new ArrayList<ScimGroupExternalMember>();
+    public int deleteByOrigin(String origin, String zoneId) {
+        // TODO Auto-generated method stub
+        return 0;
```

Comment:
Delete by origin is a bit questionable - in particular with the mapping of an origin to a domain. Generally speaking we'd only be able to delete out of the 'UAA' origin (the system tenant's system domain). I think it's better to just leave it as inoperative rather than risk blowing out everything in a tenant and rendering it unusable.